lib/datadog/appsec/processor.rb

```diff
@@ -16,9 +16,38 @@ def self.===(other)
         end
       end
 
+      # Context wraps libddwaf's context
+      class Context
+        attr_reader :time, :time_ext, :timeouts
+
+        def initialize(processor)
+          @context = Datadog::AppSec::WAF::Context.new(processor.send(:handle))
+          @time = 0.0
+          @time_ext = 0.0
+          @timeouts = 0
+        end
+
+        def run(*args)
+          start = Core::Utils::Time.get_time
+
+          ret, res = @context.run(*args)
+
+          stop = Core::Utils::Time.get_time
+
+          @time += res.total_runtime
+          @time_ext += (stop - start) * 1_000_000_000
+          @timeouts += 1 if res.timeout
```

Comment:
I suggest being explicit with the units here:

```suggestion
          start_ns = Core::Utils::Time.get_time(:nanosecond)

          ret, res = @context.run(*args)

          stop_ns = Core::Utils::Time.get_time(:nanosecond)

          @time_ns += res.total_runtime # <-- I'm not even sure what unit this is in :(
          @time_ext_ns += (stop_ns - start_ns)
          @timeouts += 1 if res.timeout
```

Reply:
`Core::Utils::Time.get_time(:nanosecond)`: Cool, I didn't knwo this was a thing!

Good point, although I like having unit-typed values+type checking instead of using hungarian notation (which is merely documentation and, although valuable, does not prevent mistakes by itself).

I need to remember to change the `attr_readers` too!

Reply:
(and yes it's nanoseconds)

Reply:
+1 I also prefer using types for this. When we're using primitives instead, I always like to annoy people to document via code :)

Reply:
Done